numba/npyufunc/dufunc.py

```diff
@@ -52,6 +63,10 @@ def __init__(self, py_func, **kws):
             kws.pop('identity', None))
         super(DUFunc, self).__init__(dispatcher, **kws)
         self._install_type()
+        self._kernel = type('DUFuncKernel_' + self.ufunc.__name__,
```

Comment:
That's a good point. My reservation is that the type() construction is more cognitive load for the reader. It takes much more time for me to parse it than a plain class declaration. I have to parse the function call as well as a dict() instantiation to know what the class actually inherits from, and overrides.